agent/xds/server.go

```diff
@@ -303,8 +308,14 @@ func (s *Server) process(stream ADSStream, reqCh <-chan *envoy.DiscoveryRequest)
 			if req.TypeUrl == "" {
 				return status.Errorf(codes.InvalidArgument, "type URL is required for ADS")
 			}
+
+			if node == nil && req.Node != nil {
+				node = req.Node
+				proxyFeatures = determineSupportedProxyFeatures(req.Node)
+			}
```

Comment:
The second thing is what happens. If you hang up and dial back again you would get a new socket and would get a new DiscoveryRequest and would recompute the proxyFeatures.

This PR also sets `"set_node_on_first_message_only": true` so that envoy only sends the Node struct on the very first DiscoveryRequest of the connection which nicely cuts down on some extra network transfer of a largish struct of static data.